enhancements/installer/openstack-customer-provided-subnets.md

```diff
@@ -0,0 +1,131 @@
+
+### Non-Goals
+There are no changes in expectations regarding publicly addressable parts of the cluster.
+
+## Proposal
+- Installer allow users to provide a list of subnets that should be used for the cluster. Since there is an expectation that networking is being shared, the installer cannot modify the networking setup, but changes required to the shared resources like Tags that do not affect the behavior for other tenants of the network will be made.
+- The installer validates the assumptions about the networking setup
+- Non-networking infrastructure resources that are specific to the cluster and resources owned by the cluster will be created by the installer. So resources like security groups, roles, RHCOS boot images, and ignition storage objects.
+- The infrastructure resources owned by the cluster continue to be clearly identifiable and distinct from other resources.
+- Destroying a cluster must make sure that no resources are deleted that didn't belong to the cluster. Leaking resources is preferred over any possibility of deleting non-cluster resources.
+
+
+## User Stories
+As an administrator of a cluster using provider networking, I have a pre-configured provider network, that cannot be manipulated by openstack tenants, and I want to install openshift on it.
+
+## Implementation Details Notes/Constraints
+
+### Installer Resources
+The user provides a list of subnets to the installer in the openstack platform section of the install-config.yaml.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  openstack:
+    region: us-west-2
+    subnets:
+    - subnet-1
+    - subnet-2
+    - subnet-3
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+### Subnets
+- Subnets must all be a part of the same OpenStack cloud
+- Each subnet must be able to be tagged, and have the capacity for at least one tag. This allows us to add the `kubernetes.io/cluster/.*:` shared tag to identify the subnet as part of the cluster. The k8s cloud provider code in kube-controller-manager uses this tag to find the subnets for the cluster to create LoadBalancer Services.
+- The CIDR block for each one must be contained by Networking.MachineCIDR.
+- Subnets must be able to reach the node that is deploying the installer, otherwise it will fail to complete the bootstrapping and installation.
+- No two public subnets or two private subnets can share a single availability zone, because that might be an error for future cloud provider load-balancer allocation.
+- Provider networks lack the concept of floating IPs, used by the installer for the VIPs for ingress and LBs. Floating IPs are reserved usually for the external network, which will have internet access and will connect a router interface to the internet, while provider networks will be L2 networks, part of the physical network infrastructure. Both provider networks and external networks are pre-created by the administrator and shared with the tenants.
```

Comment:
reading this i wasn't sure what's what. _ it might just be my lack of openstack knowledge_ so if we can maybe try to improve it that would be great